fenix/app/src/main/java/org/mozilla/fenix/home/topsites/TopSites.kt

```diff
@@ -76,6 +77,12 @@ private const val TOP_SITES_ROW_WIDTH = TOP_SITES_PER_ROW * TOP_SITES_ITEM_SIZE
 private const val TOP_SITES_FAVICON_CARD_SIZE = 60
 private const val TOP_SITES_FAVICON_SIZE = 36
 
+/**
+ * Store favicon Bitmaps fetched for a [TopSite.Provided] to prevent re-fetching and re-drawing. We
+ * use [TopSite.Provided.url] as the key as [TopSite.Provided.id] may be null.
+ */
+private val faviconBitmaps = mutableMapOf<String, Bitmap>()
+
```

Comment:
I don't really know the JVM GC strategy well enough to say for sure, but do you know how long this will stay in scope for?